### ISSUE TITLE
Picture element images: Image size issue

### ISSUE THREAD

**Issue by mukeshpanchal27** [WordPress/performance#1316]
## Bug Description

When using the new Modern Image Formats picture element support, I noticed that the markup didn't account the image sizes that are selected from editor.

## Steps to reproduce

1. Activate the Modern Image Formats plugin and enable picture support from the Media Settings screen.
2. Add an image block, Select image size to `Thumbnail`
3. Preview the page in the front end it load `Full` size image instead `Thumbnail` also the sources is missing for thumb size only. The image width and height get selected image values accordingly.

### Editor
<img width="1000" alt="Screenshot 2024-06-24 at 5 02 02 PM" src="https://github.com/WordPress/performance/assets/10103365/cb8b6f98-ada5-47b7-b7d3-5a164822d564">

### Frontend:
<img width="1346" alt="Screenshot 2024-06-24 at 5 03 26 PM" src="https://github.com/WordPress/performance/assets/10103365/d5e0873a-8b06-4e18-a33c-9067178fe333">

**Comment by imrraaj** [comment]
I can pick up this issue, @mukeshpanchal27 @westonruter can you please assign me

**Comment by imrraaj** [comment]
I am able to reproduce the bug as per the given steps. I have found the cause of the issue. When the image size is set to `Thumbnail`, it produces a sources array with a single element, which is overlooked.
https://github.com/WordPress/performance/blob/49beab84e69d05e3eaa43b8ea96d2aa8587e4142/plugins/webp-uploads/picture-element.php#L118

One simple workaround I found is that when there is a single element in the sources array, we can add an empty entry so it will be forced to show the source tag in the frontend.

```PHP
if ( count( $filtered_sources ) === 1 ) {
	$filtered_sources[] = array(
		'url'        => '',
		'descriptor' => '',
		'value'      => '',
         );
}
```
Output on the frontend: 

<img width="1605" alt="Screenshot 2024-06-26 at 2 51 06 PM" src="https://github.com/WordPress/performance/assets/71687258/99fcb5b9-de5e-4fb3-b71a-93caad37b90f">



I wanted to ask if the workaround mentioned is the best approach or if there are better solutions available. I'm open to any suggestions or alternative ideas to address this issue more effectively.

cc: @thelovekesh

**Comment by westonruter** [comment]
I think perhaps a better approach would be to change the logic around here:

https://github.com/WordPress/performance/blob/49beab84e69d05e3eaa43b8ea96d2aa8587e4142/plugins/webp-uploads/picture-element.php#L100-L102

https://github.com/WordPress/performance/blob/49beab84e69d05e3eaa43b8ea96d2aa8587e4142/plugins/webp-uploads/picture-element.php#L123-L131

If the `$sizes` is false or `$image_srcset` is empty, then `$image_srcset` should be set to the modern image format version of the `$src`.

**Comment by mukeshpanchal27** [comment]
Fixed in #1354
